classes/ActionScheduler_wpCommentCleaner.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * Class ActionScheduler_WPCommentCleaner
+ *
+ * @since 3.0.0
+ */
+class ActionScheduler_WPCommentCleaner {
+
+	/**
+	 * Post migration hook used to cleanup the WP comment table.
+	 *
+	 * @var string
+	 */
+	protected static $cleanup_hook = 'action_scheduler/cleanup_wp_comment_logs';
+
+	/**
+	 * An instance of the ActionScheduler_wpCommentLogger class to interact with the comments table.
+	 *
+	 * This instance should only be used as an interface. It should not be initialized.
+	 *
+	 * @var ActionScheduler_wpCommentLogger
+	 */
+	protected static $wp_comment_logger = null;
+
+	/**
+	 * Initialize the class and attach callbacks.
+	 */
+	public static function init() {
+		self::$wp_comment_logger = new ActionScheduler_wpCommentLogger();
```

Comment:
If it's a static property we should check whether it's already been populated before creating a new instance.

Reply:
👍 I've fixed that. :) 

While we're commenting on this line, I should raise that I'm not sure if this is the right way to go about this. 

The functions which filter the wp comments live in the `ActionScheduler_wpCommentLogger` class which is why it was necessary to get an instance of it. 

It should be fine to do this as long as the object is never initialised (`ActionScheduler_wpCommentLogger->init()`). If it's never initialised it doesn't attach its log writing callbacks. This is why I added [this comment](https://github.com/Prospress/action-scheduler/pull/369/files#diff-0f9f6c40a202f3acaabaeb0b6e9460e4R20). 

If we want to avoid creating an instance of `ActionScheduler_wpCommentLogger` though we would need to refactor those functions out which is doable of course but I don't know where they would live. We could also make them static but I think that would require us to deprecate the existing non-static ones.